src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -75,7 +75,7 @@ internal class FoldingRangeEndpoint : IVSFoldingRangeEndpoint
                 {
                     foldingRanges = await HandleCoreAsync(requestParams, documentContext, cancellationToken);
                 }
-                catch (Exception e) when (retries < MaxRetries)
+                catch (Exception e)
```

Comment:
Either `retries` etc. should be removed, or perhaps more likely we could just put the check inside the catch block, and return null?

Reply:
Actually, if we remove the "catch" entirely locally are we able to easily reproduce the "everything falls over" experience? Would make it a lot more approachable to understand why we can't handle these types of exceptions

Reply:
Are we not getting logs with these exceptions at all? I don't fully understand the system but I would expect throwing here would just log an exception later down the line than logging here.

Reply:
If we go the entire MaxRetry+1 count without having success we should probably throw our own exception / log an error. We'll want to know why throwing here breaks everything of course but we'll also want to report the failure in telemetry

Reply:
That's what the initial code was doing. It may not have been clear because I mixed two different comparison operators. 

```c#
catch (Exception e) when (retries < MaxRetries)
{

}
``` 

is basically same as 

```
catch (Exception e)
{
    if (retries == MaxRetries)
    {
		throw e;
	}
}
```

Reply:
It's not the same, because in the latter case the exception won't bubble up if the condition isn't met

Reply:
You're right though. When retries > max the second is different, which can happen in this loop. I just was my own self fulfilled prophecy of wrong